automation/check-doc.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright 2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { stripIndent } = require('common-tags');
+const _ = require('lodash');
+const { fs } = require('mz');
+const path = require('path');
+const simplegit = require('simple-git/promise');
+
+/**
+ * Compare the timestamp of cli.markdown with the timestamp of staged files,
+ * issuing an error if cli.markdown is older. Besides the purpose of ensuring
+ * that cli.markdown is updated, it effectively also ensures that coffeelint
+ * is executed (via `npm run build` or `npm test`) on the developers laptop,
+ * so that there is at least a chance that the developer will spot any linter
+ * warnings (that could reveal bugs) sooner than later.  (The CI does not
+ * currently fail in case of coffeelint warnings.)
+ * If cli.markdown does not require updating and the developer cannot run
+ * `npm run build` on their laptop, the error message suggests a workaround
+ * using `touch`.
+ */
+async function checkBuildTimestamps() {
+	const git = simplegit();
+	const docFile = path.join('doc', 'cli.markdown');
+	const [docStat, gitStatus] = await Promise.all([
+		fs.stat(docFile),
+		git.status(),
```

Comment:
`git.status()` provides similar information as running `git status` on a command line, with the result returned in the form of objects and arrays. Here it is used to identify staged files (including newly created or renamed files).

I had originally intended to use `lint-staged` (as I had suggested in some comments in PR #1589) instead of anything like `git status`, because `lint-staged`'s purpose is exactly to run scripts with staged files as arguments. Alas, I found that `lint-staged` modifies the timestamps of staged files, which completely foiled my timestamp comparison plan! No matter: `git.status()` to the rescue. It should be OK for as long as this script is executed **before** `lint-staged`.